bundles/org.openhab.binding.sonos/src/main/java/org/openhab/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -2351,20 +2386,20 @@ private void scheduleNotificationSound(final Command command) {
      * @param url in the format of //host/folder/filename.mp3
      */
     public void playNotificationSoundURI(Command notificationURL) {
-        if (notificationURL != null && notificationURL instanceof StringType) {
+        if (notificationURL instanceof StringType) {
             try {
                 ZonePlayerHandler coordinator = getCoordinatorHandler();
 
                 String currentURI = coordinator.getCurrentURI();
                 logger.debug("playNotificationSoundURI: currentURI {} metadata {}", currentURI,
                         coordinator.getCurrentURIMetadataAsString());
 
-                if (isPlayingStream(currentURI) || isPlayingRadioStartedByAmazonEcho(currentURI)
-                        || isPlayingRadio(currentURI)) {
+                if ((currentURI != null) && (isPlayingStream(currentURI)
+                        || isPlayingRadioStartedByAmazonEcho(currentURI) || isPlayingRadio(currentURI))) {
                     handleRadioStream(currentURI, notificationURL, coordinator);
-                } else if (isPlayingLineIn(currentURI)) {
+                } else if (currentURI != null && isPlayingLineIn(currentURI)) {
                     handleLineIn(currentURI, notificationURL, coordinator);
-                } else if (isPlayingQueue(currentURI)) {
+                } else if (currentURI != null && isPlayingQueue(currentURI)) {
```

Comment:
The null check already done in the `isPlayingQueue` method, so you don't need to check beforehand.
```suggestion
                } else if (isPlayingQueue(currentURI)) {
```

Reply:
Yes you're right, I have to check why I added that, I just remember it was to avoid a warning.

Reply:
Ok, it was in fact to avoid an error at the next line (call to handleSharedQueue with a nullable currentURI).

Reply:
Fixed